go/vt/vtctl/vtctl.go

```diff
@@ -3644,6 +3650,70 @@ func commandGetSrvKeyspace(ctx context.Context, wr *wrangler.Wrangler, subFlags
 	return printJSON(wr.Logger(), cellKs)
 }
 
+func commandUpdateThrottlerConfig(ctx context.Context, wr *wrangler.Wrangler, subFlags *pflag.FlagSet, args []string) (err error) {
+	noValueIndicator := "~"
+	enable := subFlags.Bool("enable", false, "Enable the throttler")
+	disable := subFlags.Bool("disable", false, "Disable the throttler")
+	threshold := subFlags.Float64("threshold", 0, "threshold for the either default check (replication lag seconds) or custom check")
+	customQuery := subFlags.String("custom_query", noValueIndicator, "custom throttler check query")
+	checkAsCheckSelf := subFlags.Bool("check_as_check_self", false, "/throttler/check requests behave as is /throttler/check-self was called")
+	checkAsCheckShard := subFlags.Bool("check_as_check_shard", false, "use standard behavior for /throttler/check requests")
+
+	if err := subFlags.Parse(args); err != nil {
+		return err
+	}
+	if subFlags.NArg() != 1 {
+		return fmt.Errorf("the <keyspace> arguments are required for the SetThrottlerConfig command")
+	}
+	if *enable && *disable {
+		return fmt.Errorf("--enable and --disable are mutually exclusive")
+	}
+
+	keyspace := subFlags.Arg(0)
+
+	update := func(throttlerConfig *topodatapb.SrvKeyspace_ThrottlerConfig) *topodatapb.SrvKeyspace_ThrottlerConfig {
+		if throttlerConfig == nil {
+			throttlerConfig = &topodatapb.SrvKeyspace_ThrottlerConfig{}
+		}
+		if *customQuery != noValueIndicator {
+			// custom query provided
+			throttlerConfig.CustomQuery = *customQuery
+			throttlerConfig.Threshold = *threshold // allowed to be zero/negative because who knows what kind of custom query this is
+		} else {
+			// no custom query, throttler works by querying replication lag. We only allow positive values
+			if *threshold > 0 {
+				throttlerConfig.Threshold = *threshold
+			}
+		}
+		if *enable {
+			throttlerConfig.Enabled = true
+		}
+		if *disable {
+			throttlerConfig.Enabled = false
+		}
+		if *checkAsCheckSelf {
+			throttlerConfig.CheckAsCheckSelf = true
+		}
+		if *checkAsCheckShard {
+			throttlerConfig.CheckAsCheckSelf = false
+		}
+		return throttlerConfig
+	}
+
+	ctx, unlock, lockErr := wr.TopoServer().LockKeyspace(ctx, keyspace, "UpdateThrottlerConfig")
+	if lockErr != nil {
+		return lockErr
+	}
+	defer unlock(&err)
+
+	updatedCells, err := wr.TopoServer().UpdateSrvKeyspaceThrottlerConfig(ctx, keyspace, []string{}, update)
+	if err != nil {
+		return err
+	}
```

Comment:
we return the list of affected cells, if that makes sense.

Reply:
We should make it clear to the user if the affected cells != all cells, alerting them that they may need to run the command again, no? I didn't see where we allow them to specify cells so it should always be all. For a command that doesn't have a cells flag, this output is not obvious and it could be easy to miss the fact that e.g. one of your eight zones is not listed — what that means and what action to take.

Reply:
I believe this is handled in:

https://github.com/vitessio/vitess/pull/11604/files#diff-41eef6b3446b49f65c87aef05983e5012b66eaa61a0a9fada2da18bfae7056ccR436-R438

```
	if rec.HasErrors() {
		return cells, NewError(PartialResult, rec.Error().Error())
	}
```

The user _does_ get told if any of the cells failed.

Reply:
I should note this behavior is consistent with all other srv_keyspace functions; they all do this same check and return the same `PartialResult`